src/lib/safe/signature-verification/SignatureVerifierStrategy.ts

```diff
@@ -0,0 +1,83 @@
+import Safe from "@safe-global/protocol-kit";
+import { ethers } from "ethers";
+
+const EIP1271_MAGIC_VALUE =
+  "0x1626ba7e00000000000000000000000000000000000000000000000000000000";
+
+export interface SignatureVerifierStrategy {
+  verify(
+    signature: `0x${string}`,
+    rpcUrl: string,
+    safeAddress: `0x${string}`,
+  ): Promise<boolean>;
+}
+
+export class SignatureVerifierStrategyFactory {
+  static getStrategy(
```

Comment:
Since choosing the strategy isn't influenced by the hash of the typed data, I feel like it shouldn't be here. It also doesn't necessarily need to be on the constructor if the other fields aren't either. From the viewpoint of the individual classes you can't necessarily say that the hash is a value that you would want to keep across multiple invocations, so the actual call to `verify()` seems like it's lacking this value.

It's absolutely fine the way it is, but would probably be something that I'd refactor given I have the time and am working on something that's touching this code :)